src/algebraic_topology/alternating_face_map_complex.lean

```diff
@@ -0,0 +1,233 @@
+      simp only [fin.cast_succ_cast_lt] at h1,
+      exact h1, }, },
+  { -- φ : S → Sᶜ is surjective
+    clear term d_l d_r,
+    rintro ⟨i', j'⟩ hij',
+    simp only [true_and, finset.mem_univ, finset.compl_filter, not_le,
+      finset.mem_filter] at hij',
+    refine ⟨(j'.pred _,fin.cast_succ i'),_,_⟩,
+    { intro H,
+      rw [H] at hij',
+      simp only [nat.not_lt_zero, fin.coe_zero] at hij',
+      exact hij', },
+    { simp only [true_and, finset.mem_univ, fin.coe_cast_succ, fin.coe_pred,
+        finset.mem_filter],
+      exact nat.le_pred_of_lt hij', },
+    { simp only [prod.mk.inj_iff, fin.succ_pred, fin.cast_lt_cast_succ],
+      split; refl, }, },
+end
+
+/-!
+## Construction of the alternating face map complex functor
+-/
+
+/-- The alternating face map complex, on objects -/
+def obj : chain_complex C ℕ := chain_complex.of (λ n, X _[n]) (obj_d X) (d_squared X)
+
+variables {X} {Y}
+
+/-- The alternating face map complex, on morphisms -/
+@[simp]
+def map (f : X ⟶ Y) : obj X ⟶ obj Y :=
+chain_complex.of_hom _ _ _ _ _ _
+  (λ n, f.app (op [n]))
+  (λ n,
+    begin
+      repeat { rw obj_d },
+      rw [comp_sum, sum_comp],
+      apply congr_arg,
+      ext,
+      rw [comp_zsmul, zsmul_comp],
+      apply congr_arg,
+      erw f.naturality,
+      refl,
+    end)
+
+end alternating_face_map_complex
+
+variables (C : Type*) [category C] [preadditive C]
+
+/-- The alternating face map complex, as a functor -/
+@[simps]
+def alternating_face_map_complex : simplicial_object C ⥤ chain_complex C ℕ :=
+{ obj := alternating_face_map_complex.obj,
+  map := λ X Y f, alternating_face_map_complex.map f }
+
+/-!
+## Construction of the natural inclusion of the normalized Moore complex
+-/
+
+variables {A : Type*} [category A] [abelian A]
+
+/-- The inclusion map of the Moore complex in the alternating face map complex -/
+def inclusion_of_Moore_complex_map (X : simplicial_object A) :
+  (normalized_Moore_complex A).obj X ⟶ (alternating_face_map_complex A).obj X :=
+chain_complex.of_hom _ _ _ _ _ _
+  (λ n, (normalized_Moore_complex.obj_X X n).arrow)
+  (λ n,
+    begin
+      /- we have to show the compatibility of the differentials on the alternating
+         face map complex with those defined on the normalized Moore complex:
+         we first get rid of the terms of the alternating sum that are obviously
+         zero on the normalized_Moore_complex -/
+      simp only [alternating_face_map_complex.obj_d],
+      rw comp_sum,
+      let t := λ (j : fin (n+2)), (normalized_Moore_complex.obj_X X (n+1)).arrow ≫
+        ((-1 : ℤ)^(j : ℕ) • X.δ j),
+      have def_t : (∀ j : fin (n+2), t j = (normalized_Moore_complex.obj_X X (n+1)).arrow ≫
+        ((-1 : ℤ)^(j : ℕ) • X.δ j)) := by { intro j, refl, },
+      rw [fin.sum_univ_succ t],
+      have null : ∀ j : fin (n+1), t j.succ = 0,
+      { intro j,
+        rw def_t,
+        rw comp_zsmul,
+        rw ← zsmul_zero ((-1 : ℤ)^(j.succ : ℕ)),
+        apply congr_arg,
```

Comment:
It's perfectly fine to leave `apply congr_arg` here (and a few other places), but you might want to look at the `congr` and `congr'` tactics as well.

Reply:
Thanks for all the suggestion, which I have applied.